verilog/CST/functions.cc

```diff
@@ -50,12 +50,65 @@ std::vector<verible::TreeSearchMatch> FindAllFunctionHeaders(
 
 std::vector<verible::TreeSearchMatch> FindAllFunctionOrTaskCalls(
     const Symbol& root) {
-  return verible::SearchSyntaxTree(root, NodekFunctionCall());
+  std::vector<verible::TreeSearchMatch> names;
```

Comment:
I'm generally a little concerned when a `Find*` function becomes non-trivial.

As an alternative to detecting one of multiple tree patterns, is it possible to disambiguate the syntax as it is being constructed from the bottom-up?  By this, I mean that you might have a grammar rule that is without grammar conflict, but still have one of several semantic interpretations, say "A-or-B", but as the rule gets reduced, it becomes clear whether it is an "A" or "B".  When it becomes clear, you can rewrite the subtree that is already formed, so the transient "A-or-B" designation is erased.  We can say that a syntax tree shall not be searched until it is fully formed.

Reply:
I think simplified it to be in fairly unambiguous nodes - I have managed to dodge a lot of ambiguity, edge cases and S/R-s while containing everything that looks like a call to be a function call (anonymous single declarations will be treated as function calls - unavoidable without a symbol table reference telling that eg. foo is a data type not a function when dealing with a line like `foo ();` or `foo (args);` - in some cases they can be either a declaration or a function call in the same exact place in code). It's back to original form now.